Add optional testing for Python 3.15

avylove opened the pull request:
Added initial testing for 3.15-rc. Most of the tests are failing for.
```python
DeprecationWarning: This process (pid=238) is multi-threaded, use of forkpty() may lead to deadlocks in the child.
```

There is an entry about this in the [docs](https://docs.python.org/3.15/library/os.html#os.fork) that references a [discussion thread](https://discuss.python.org/t/concerns-regarding-deprecation-of-fork-with-alive-threads/33555).

I haven't dug into this because I don't have a system package for 3.15 yet. One option is to ignore the warning because this is testing and it has worked fine for years. Just figured it would be good to have it on the radar.

jquast commented:
Sure, just ignore it within the as_subprocess context manager looks like it would take care of it all, my pending PR's have a new pty helper that it would also be ignored in that covers the manual uses of pty.fork. It is safe to ignore, we are not calling urllib/ssl etc.

avylove commented:
That fixed most of them. Two cases of test_SequenceWrapper() are still failing. I need to dig in more, but wondering if it's related to https://github.com/python/cpython/commit/1c598e04361dbfc9cf465f3a02f83715c11b028c since it's the only change since 3.14.

jquast commented:
It does look from the test this is the cause, we should adapt the fix into our own wrap() whether or not it is, unfortunately our wrap method is duplicated from upstream and this test is part of ensuring it behaves the same. This will "invert" the test results, failing for versions 3.13 and under.

From there, I think it is best for code coverage to do a "fixup" for versions under 3.14, the standard library wrap result is modified for the extra ``' '`` item, only for versions and parametrized test combinations affected. This should cause all tests to pass on all python versions and still retain full coverage

avylove commented:
It looks like the new behavior will stay and it's currently expected to be in 3.13 and 3.14 when they do their next point release.

I updated most of `` _wrap_chunks()``  to match the 3.15 behavior, then added a workaround for older versions. It currently looks for 3.15+ specifically, so will need to be updated when 3.13 and 3.14 are updated. Another option is to generalize it and implement the workaround any time the first line is only whitespace. Thoughts?

Since the new behavior isn't in any released versions, it probably makes sense to old off on this until 3.13 and 3.14 are updated, or at least 3.14. This should only affect cases where the preceding whitespace is longer than the width.

The updated logic should now handle the `max_lines` and  `placeholder` options, so I added a keyword combinations for that in the tests.

There are some uncovered lines in the tests. I need to look at that. It probably needs a new test rather than updating the keyword args.

``_handle_long_words()`` got a minor change, but I might take another look and see if I can get ``break_on_hyphens`` to work
